src/data-api.js

```diff
@@ -80,7 +80,13 @@ function dataAPI(ClientRDSDataAPI, Client, dialect) {
     },
 
     // Process the response as returned from the query, and format like the standard mysql engine
-    processResponse(obj) {
+    processResponse(obj, runner) {
+      if (obj == null) return;
```

Comment:
This change emulates the mysql dialect compiler in knex.

See here: https://github.com/knex/knex/blob/7cfb78d16aa9dcd635c9ae1862837ccd4f507a21/lib/dialects/mysql/index.js#L149